chain/epoch-manager/src/adapter.rs

```diff
@@ -475,6 +481,11 @@ impl EpochManagerAdapter for EpochManagerHandle {
         epoch_manager.is_next_block_epoch_start(parent_hash)
     }
 
+    fn is_last_block_in_finished_epoch(&self, hash: &CryptoHash) -> Result<bool, EpochError> {
+        let epoch_manager = self.read();
```

Comment:
What does this do?

Reply:
This is `impl EpochManagerAdapter for EpochManagerHandle`
https://github.com/near/nearcore/blob/fdead431c9bdafc105d12afd272d50066261feff/chain/epoch-manager/src/lib.rs#L61

The actual implementation of the method depends on specific `EpochManager` implementation of `get_epoch_info`. So we get `EpochManager` out of `EpochManagerHandle` and call `is_last_block_in_finished_epoch` there.